feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/TimetableItemDetailViewModel.kt

```diff
@@ -60,6 +62,16 @@ class TimetableItemDetailViewModel @Inject constructor(
     fun onBookmarkClick(timetableItem: TimetableItem) {
         viewModelScope.launch {
             sessionsRepository.toggleBookmark(timetableItem.id)
+            val bookmarked = timetableItemStateFlow.value?.second ?: return@launch
+            if (bookmarked) {
+                userMessageStateHolder.showMessage(
+                    TimetableItemDetailStrings.BookmarkedSuccessfully.asString(),
+                    TimetableItemDetailStrings.ViewBookmarkList.asString(),
+                )
+            } else {
+                val currentMessage = userMessageStateHolder.messageUiState.userMessages.firstOrNull() ?: return@launch
+                userMessageStateHolder.messageShown(currentMessage.id, Dismissed)
```

Comment:
When the button is tapped, the Snackbar disappears, but at that time, the Dissmiss event does not occur, so I manually dismiss it 😢

Reply:
"The button" means "View" button or Bookmark button?

Reply:
Bookmark button.
When the bookmark button is tapped, the Snackbar disappears, but it seems that `userMessageStateHolder.messageShown` is not being called. 
Do you have any idea why this might be happening?

Reply:
I tried, but it was the same userMessage.

```kotlin
    userMessageStateHolder.messageUiState.userMessages.firstOrNull()?.let { userMessage ->
        println(userMessageStateHolder.messageUiState.userMessages) // ← added
        LaunchedEffect(userMessage) {
```

first: 
```
[UserMessage(message=ブックマークに追加されました, actionLabel=一覧を見る, id=-5097040755527889007, userMessageResult=null)]
```

second: 
```
[UserMessage(message=ブックマークに追加されました, actionLabel=一覧を見る, id=-5097040755527889007, userMessageResult=null), UserMessage(message=ブックマークに追加されました, actionLabel=一覧を見る, id=2112774890980393949, userMessageResult=null)]
```

In this pattern, it seems that `snackbarHostState.showSnackbar` is not returning `SnackbarResult`. As a result, the suspend function in the `filter` gets stuck, and `userMessages` is not removed.

https://github.com/DroidKaigi/conference-app-2023/blob/86faf75d8b35cc190e912b7230cf105ec47ab497/core/ui/src/androidMain/kotlin/io/github/droidkaigi/confsched2023/ui/UserMessage.kt#L73-L89

 This causes `firstOrNull` to yield the same result.

Reply:
Can you dive into snackbarHostState.showSnackbar? Or can you provide a reproducing branch? 🙏

Reply:
OK. I will give it a try, so please wait for a while.（or use this branch https://github.com/KoutaMatsushita/conference-app-2023/tree/bookmark_confirmation_snackbar ）

Reply:
SnackbarHost is disappearing due to recompose. 😓 
suppress recompose using `remember { SnackbarHostState() }`.

Reply:
I'll keep that in mind 😭 Thanks for investigating!